src/diagram/node_registry.rs

```diff
@@ -469,17 +464,20 @@ pub trait IntoNodeRegistration {
         self,
         id: BuilderId,
         name: String,
-        gen: &mut SchemaGenerator,
+        schema_generator: &mut SchemaGenerator,
     ) -> NodeRegistration;
 }
 
 pub struct NodeRegistry {
     nodes: HashMap<BuilderId, NodeRegistration>,
+    pub(super) data: DataRegistry,
+}
 
+pub struct DataRegistry {
```

Comment:
I'm inclined to avoid calling it `DataRegistry`, "Data" is too generic. Maybe `CommonOperations` can be renamed to `CommonOperationsBuilder`, then this can be `CommonOperations`.

Reply:
I named it "data" to refer to the data that moves between nodes. I would actually recommend that we eventually move the impls for `unzip`, `split`, and all the rest into here while `NodeRegistration` only contains a `TypeId` for `Request` and `Response`. That would allow us to have one central database of all operations that can be done on each data type. Then we can register operations for known data types separately from registering node builders. However this may conflict with the node metadata, so we'd need to reconsider how that's presented.

In any case, all of the above ideas can be saved for a future PR.

What if we named this `MessageRegistry` instead of `DataRegistry`?